### PR TITLE
Basic isolated example for rules_cc.

### LINKED ISSUE #15: separate examples and downloadable repository
Currently all the things below are intermixed in a flat-ish hierarchy of bazel-zig-cc:
- files that are designed to be loaded by third parties: `WORKSPACE` and `toolchain/`.
- repository bookkeeping (gazelle, lint, release scripts)
- `tests/`.

The original idea was to have the "minimal required things" be in the top level of the repository; meaning the `WORKSPACE` file would have only such contents: `workspace(name = "bazel-zig-cc")`; everything else should be hidden in sub-directories. However, I can't figure out a way to make bookeeping work -- which depends on Gazelle, Go and all that fun stuff. Basically, it's a conflict of what should be in the `WORKSPACE`.

Concretely:
1. Could the "book-keeping" `WORKSPACE` be in it's sub-folder, and the bookkeeping commands be run from there? To be clear, we want the bookkeeping commands to access the full repository.
2. Can this be made easier with bzlMod? Looking for bzlMod experts here.

### PR DESCRIPTION
**Pull request by laurynaslubys**
Add a basic isolated examples for rules_cc, as a starting point for users.

Fixes #15.

### CODE REVIEW

File: ci/test_example [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
```
Comment (motiejus) [inline ci/test_example:1]: Let's use `/bin/sh` for simple scripts.
```suggestion
#!/bin/sh
```

File: examples/rules_cc/WORKSPACE [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        "https://github.com/uber/hermetic_cc_toolchain/releases/download/{0}/hermetic_cc_toolchain-{0}.tar.gz".format(HERMETIC_CC_TOOLCHAIN_VERSION),
+    ],
+)
+
+load("@hermetic_cc_toolchain//toolchain:defs.bzl", zig_toolchains = "toolchains")
+
+zig_toolchains()
+
+register_toolchains(
+    "@zig_sdk//toolchain:linux_amd64_gnu.2.19",
```
Comment (motiejus) [inline examples/rules_cc/WORKSPACE:22]:
```suggestion
    "@zig_sdk//toolchain:linux_amd64_gnu.2.28",
```

File: examples/rules_cc/hello_world_test.cc [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    int result = std::system(command.c_str());
+
+    if (result == 0) {
+        std::cout << "Hello, World! test passed." << std::endl;
+    } else {
+        std::cerr << "Hello, World! test failed. Return code: " << result << std::endl;
+        return 1;
+    }
+
+    return 0;
```
Comment (motiejus) [inline examples/rules_cc/hello_world_test.cc:23]:
```suggestion
    return std::system(command.c_str());
```

how about the previous suggestion?

File: examples/rules_cc/WORKSPACE (hunk 2 above)
Comment (motiejus) [reply examples/rules_cc/WORKSPACE:22]: Let's use an up-to-date libc in examples. Users can downgrade if they need it themselves.

**Review by linzhp — APPROVED** [review]

File: examples/rules_cc/hello_world_test.cc (hunk 3 above)
Comment (laurynasl-uber) [reply examples/rules_cc/hello_world_test.cc]: I just replaced this with a `cc_test`.

**Review by motiejus — APPROVED** [review]

**Review by motiejus — APPROVED** [review]

**Pull request merged**